ambry-account/src/main/java/com/github/ambry/account/mysql/MySqlDataAccessor.java

```diff
@@ -143,11 +153,34 @@ public static boolean isCredentialError(SQLException e) {
   /**
    * Handle a SQL exception on a database operation.
    * @param e the {@link SQLException} encountered.
+   * @param operationType type of mysql operation
    */
-  void onException(SQLException e) {
+  void onException(SQLException e, OperationType operationType) {
     if (e instanceof SQLTransientConnectionException) {
       reset();
     }
+    if (!(e instanceof SQLIntegrityConstraintViolationException || isCredentialError(e))) {
```

Comment:
It might suffice to check if e instanceof SQLTransientConnectionException as above.  Are there cases that wouldn't handle?